tests/suite/system_suite_test.go

```diff
@@ -187,8 +189,13 @@ func setup(cfg setupConfig, extraInstallArgs ...string) {
 	output, err := framework.InstallGatewayAPI(cfg.gwAPIVersion)
 	Expect(err).ToNot(HaveOccurred(), string(output))
 
-	output, err = framework.InstallNGF(installCfg, extraInstallArgs...)
-	Expect(err).ToNot(HaveOccurred(), string(output))
+	if cfg.infoLogLevel {
```

Comment:
We don't really need two functions to install. We could just do:

```
if cfg.debug {
    	extraInstallArgs = append(
		    extraInstallArgs, 
		    "--set", "nginxGateway.config.logging.level=debug",
	    )
}

output, err = framework.InstallNGF(installCfg, extraInstallArgs...)
Expect(err).ToNot(HaveOccurred(), string(output))

```

Reply:
I thought about doing it that way, then also thought about the "best practice" of not letting a function have multiple purposes. In this case I guess the function already has many purposes because of the extraInstallArgs, but this felt very boolean like. 

I can make the change if you think this is a case where we are fine.

Reply:
In this case, we are still only doing one thing—installing NGF. If we passed the boolean to InstallNGF and forced it to decide based on it, we would be violating the single responsibility principle.

The InstallNGF function is written well because it is extensible. When we need to set a new helm value, we don't have to modify InstallNGF. 

We could create helper functions like `InstallNGFDebug` that are just wrappers around `InstallNGF` and set the appropriate extraArgs. That would be helpful for common installation combinations used by more than one test.

Reply:
> The InstallNGF function is written well because it is extensible. When we need to set a new helm value, we don't have to modify InstallNGF.

Ah okay this explanation is very helpful, thanks!